### PR TITLE
Add count services, clients & test

### PR DESCRIPTION
**Pull request by leeminju531**
this PR is for service counts using it's name like topic.
Related to https://github.com/ros2/ros2cli/pull/771

Signed-off-by: leeminju531 <dlalswn531@naver.com>

### CODE REVIEW

File: rclpy/test/test_node.py [hunk 1]
```diff
@@ -337,6 +337,41 @@ def test_count_publishers_subscribers(self):
...
+        self.assertEqual(1, self.node.count_clients(short_service_name))
+        self.assertEqual(1, self.node.count_clients(fq_service_name))
+        self.assertEqual(1, self.node.count_services(short_service_name))
+        self.assertEqual(1, self.node.count_services(fq_service_name))
+
+        self.node.create_client(GetParameters, short_service_name)
+        self.assertEqual(2, self.node.count_clients(short_service_name))
+        self.assertEqual(2, self.node.count_clients(fq_service_name))
+        self.assertEqual(1, self.node.count_services(short_service_name))
+        self.assertEqual(1, self.node.count_services(fq_service_name))
```
Comment (fujitatomoya) [inline rclpy/test/test_node.py:363]: what about creating multiple server case?

**Review by fujitatomoya — APPROVED** [review]
lgtm with minor comment.

File: rclpy/test/test_node.py (hunk 1 above)
Comment (leeminju531) [reply rclpy/test/test_node.py:363]: added. sorry for late :)

**Review by clalancette — APPROVED** [review]
This looks good to me, but may need to be rebased to pass CI.

**Comment by leeminju531** [comment]
Cool. rebased:thumbsup:.

**Comment by clalancette** [comment]
CI for this is in https://github.com/ros2/rclcpp/pull/2072#issuecomment-1751784410

**Pull request merged**